### PR TITLE
GRASP-12267 load the configuration before starting the tests

### PR DESCRIPTION
**Pull request by krzyfre**
[https://nomagic.atlassian.net/browse/GRASP-12267](https://nomagic.atlassian.net/browse/GRASP-12267)

Review: @michalvi @mkatliar 

###How was the change tested?
On virtual robot through script

### CODE REVIEW

File: src/v1_0/rw/cfg.cpp [hunk 1]
```diff
@@ -5,12 +5,12 @@ namespace abb :: rws :: v1_0 :: rw :: cfg
     void loadCFGFile(RWSClient& client, const FileResource& resource)
     {
         std::stringstream uri;
-        uri << Resources::RW_CFG << "/action=load";
+        uri << Resources::RW_CFG << "?action=load";
```
Comment (mkatliar) [inline src/v1_0/rw/cfg.cpp:8]: How did it work before?

File: src/v1_0/rw/ctrl.cpp [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
     {
         std::stringstream uri;
         uri << Services::CTRL;
 
-        std::string content = "restart-mode=restart";
+        std::stringstream content;
+        content << "restart-mode=" << restartMode;
 
-        client.httpPost(uri.str(), content);
+        client.httpPost(uri.str(), content.str(), std::set<Poco::Net::HTTPResponse::HTTPStatus> {Poco::Net::HTTPResponse::HTTP_NO_CONTENT, Poco::Net::HTTPResponse::HTTP_OK});
```
Comment (mkatliar) [inline src/v1_0/rw/ctrl.cpp:13]: Shorter:
```suggestion
        client.httpPost(uri.str(), content.str(), {Poco::Net::HTTPResponse::HTTP_NO_CONTENT, Poco::Net::HTTPResponse::HTTP_OK});
```

File: include/abb_librws/common/rw/ctrl.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+namespace abb :: rws :: rw
+{
+     /**
+     * @brief Controller restart type.
+     *
+     * The documentation strings are taken from https://developercenter.robotstudio.com/api/rwsApi/ctrl_restart_page.html
+     */
+    enum class RestartMode
+    {
+        // The controller will be restarted. The state is saved and any changed system parameter settings will be activated after the restart.
```
Comment (mkatliar) [inline include/abb_librws/common/rw/ctrl.h:17]: Please use Doxygen-style comments

File: include/abb_librws/v1_0/rw/ctrl.h [hunk 4]
```diff
@@ -22,6 +23,6 @@ namespace abb :: rws :: v1_0 :: rw :: ctrl
      *
      * \throw \a RWSError if something goes wrong.
      */
-    void restartController(RWSClient& client);
+    void restartController(RWSClient& client, RestartMode const& restartMode = RestartMode::restart);
```
Comment (mkatliar) [inline include/abb_librws/v1_0/rw/ctrl.h:26]: Please document all parameters

File: include/abb_librws/v2_0/rw/ctrl.h [hunk 5]
```diff
@@ -23,6 +24,5 @@ namespace abb :: rws :: v2_0 :: rw :: ctrl
      *
      * \throw \a RWSError if something goes wrong.
      */
-    void restartController(RWSClient& client, Mastership const& mastership);
-
+    void restartController(RWSClient& client, Mastership const& mastership, RestartMode const& restartMode = RestartMode::restart);
```
Comment (mkatliar) [inline include/abb_librws/v2_0/rw/ctrl.h:27]: Please document all parameters

File: src/v1_0/rw/ctrl.cpp (hunk 2 above)
Comment (mkatliar) [inline src/v1_0/rw/ctrl.cpp:13]: Why two codes are accepted?

**Review by mkatliar — CHANGES_REQUESTED** [review]
Looks good, except a few things that I have commented on.

File: src/v1_0/rw/cfg.cpp (hunk 1 above)
Comment (krzyfre) [reply src/v1_0/rw/cfg.cpp:8]: No, it didn't work, that's why I changed it

File: src/v1_0/rw/ctrl.cpp (hunk 2 above)
Comment (krzyfre) [reply src/v1_0/rw/ctrl.cpp:13]: When I tested it, my controller returned HTTP_OK, but according to RWS1 documentation it says both can be returned, so I'm not sure whether on a different RW6 version a different code cannot be returned. Since both are of the success group I kept both

Comment (mkatliar) [reply src/v1_0/rw/ctrl.cpp]: Makes sense. Could you point me to the place where it says both codes can be returned?

File: src/v1_0/rw/cfg.cpp [hunk 6]
```diff
@@ -5,12 +5,12 @@ namespace abb :: rws :: v1_0 :: rw :: cfg
...
 
         // Path to file should be a direct path, i.e. without "/fileservice/"
         std::string content =
             "filepath=" + resource.directory + "/" + resource.filename + "&action-type=replace";
 
-        client.httpPost(uri.str(), content);
+        client.httpPost(uri.str(), content, std::set<Poco::Net::HTTPResponse::HTTPStatus> {Poco::Net::HTTPResponse::HTTP_NO_CONTENT, Poco::Net::HTTPResponse::HTTP_OK});
```
Comment (mkatliar) [inline src/v1_0/rw/cfg.cpp:14]: Why did you change this? Do you get different response codes?

**Review by mkatliar — CHANGES_REQUESTED** [review]
Looks good to me in general, although I am a concerned about accepting several HTTP responses instead of a single one. I leave it at your discretion, but if you go for accepting multiple return codes, please add a comment in the code why you do this.

**Review by mkatliar — APPROVED** [review]

**Review by mkatliar — APPROVED** [review]
Looks good! Please squash the commits when merging.
